src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+                                                   ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE };
+
+// clang-format off
+static s32 sVisualStateTable[] = {
+    0, EN_DT_ANIMATION_1, EN_DT_EYE_TEXTURE_LOOK_DOWN, true,  
+    0, EN_DT_ANIMATION_2, EN_DT_EYE_TEXTURE_LOOK_DOWN, true,
+    0, EN_DT_ANIMATION_3, EN_DT_EYE_TEXTURE_LOOK_DOWN, true,  
+    0, EN_DT_ANIMATION_3, EN_DT_EYE_TEXTURE_SHOCK,     false,
+    0, EN_DT_ANIMATION_5, EN_DT_EYE_TEXTURE_SHOCK,     false, 
+    0, EN_DT_ANIMATION_4, EN_DT_EYE_TEXTURE_SHOCK,     false
+};
+// clang-format on
+
+static TexturePtr sEyeTextures[] = { gDotourEyeShockTex, gDotourEyeOpenTex, gDotourEyeClosedTex, gDotourEyeLookDownTex,
+                                     gDotourEyeSquintTex };
+
+static TexturePtr sBrowTextures[] = { gDotourEyebrowHighTex, gDotourEyebrowMidTex, gDotourEyebrowLowTex };
+
+void EnDt_Init(Actor* thisx, PlayState* play) {
+    EnDt* this = THIS;
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 19.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &object_dt_Skel_00B0CC, &object_dt_Anim_00112C, this->jointTable,
+                       this->morphTable, OBJECT_DT_LIMB_MAX);
+
+    this->actor.targetMode = TARGET_MODE_6;
+    this->npcEnMuto = NULL;
+    this->npcEnBaisen = NULL;
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    if ((gSaveContext.save.day == 3) && (gSaveContext.save.isNight)) {
+        EnDt_SetupFinalNightState(this, play);
+    } else {
+        s32 csId = this->actor.csId;
+        s32 i = 0;
+
+        // clang-format off
+        while (csId != CS_ID_NONE) { this->csIds[i] = csId; csId = CutsceneManager_GetAdditionalCsId(csId); i++; }
+        // clang-format on
+    }
+    this->actionFunc = EnDt_SetupCutsceneNpcs;
+}
+
+void EnDt_Destroy(Actor* thisx, PlayState* play) {
+    EnDt* this = THIS;
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnDt_UpdateHeadRotate(EnDt* this) {
+    s32 deltaAngle = BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.world.rot.y);
+    s32 absDeltaAngle = ABS_ALT(deltaAngle);
+
+    this->headRotTarget.y = 0;
+    if (absDeltaAngle < 20000) {
+        this->headRotTarget.y = BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.world.rot.y);
+
+        if (this->headRotTarget.y > 10000) {
+            this->headRotTarget.y = 10000;
+        } else if (this->headRotTarget.y < -10000) {
+            this->headRotTarget.y = -10000;
+        }
+    }
+}
+
+void EnDt_ChangeAnim(EnDt* this, s32 animIndex) {
+    f32 morphFrames;
+
+    this->animIndex = animIndex;
+
+    morphFrames = -4.0f;
+    if (this->animIndex == EN_DT_ANIMATION_2 || this->animIndex == EN_DT_ANIMATION_5) {
+        morphFrames = 0.0f;
+    }
+
+    this->animEndFrame = (f32)Animation_GetLastFrame(sAnimations[this->animIndex]);
```

Comment:
Is the cast here needed for matching?

Reply:
Not actually needed, leftover from m2c. Good catch.